packages/elements/src/hooks/use-media-query/index.tsx

```diff
@@ -1,30 +1,52 @@
-import { useEffect, useState } from 'react'
+import React, { createContext, useContext, useEffect, useState } from 'react'
 
 export interface MediaType {
   isMobile: boolean
   isTablet: boolean
   isDesktop: boolean
+  isWideScreen: boolean
 }
 
 export const MOBILE_BREAKPOINT = 768
 export const TABLET_BREAKPOINT = 1024
+export const DESKTOP_BREAKPOINT = 1400
 
-export const useMediaQuery = (): MediaType => {
-  const [windowWidth, setWindowWidth] = useState(() => window.innerWidth)
+export const MediaStateContext = createContext<MediaType>({} as MediaType)
+
+const { Provider } = MediaStateContext
+
+export const MediaStateProvider: React.FC = ({ children }) => {
+  const [mediaType, setMediaType] = useState<MediaType>({
+    isMobile: true,
+    isTablet: false,
+    isDesktop: false,
+    isWideScreen: false,
+  })
 
   useEffect(() => {
-    const onResize = () => setWindowWidth(window.innerWidth)
+    const onResize = () => {
+      const windowWidth = window.innerWidth
+      const isMobile = windowWidth < 768
+      const isTablet = windowWidth >= 768 && windowWidth < 1024
+      const isDesktop = windowWidth >= 1024 && windowWidth < 1400
+      const isWideScreen = windowWidth >= 1400
```

Comment:
again can these be constants please? :p you may want to change them in the future or describe them